pkg/kubernetes/scripts/policy.js

```diff
@@ -174,6 +177,66 @@
                 return result;
             }
 
+            var sarCache = { };
+
+            function subjectAccessReview(project, user, verb, resource) {
+                var key = (project || "") + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+                // cached already?
+                if (key in sarCache)
+                    return sarCache[key];
+
+                // pre-fill cache to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                loader.watch("Projects").then(function() {
+                    // if no project was given, check the first one as an approximation
+                    // TODO: is that sufficient, or should we iterate over all
+                    // projects? (much more expensive)
+                    if (user && !project) {
+                        var projects = select().kind("Project");
+                        for (var p in projects) {
+                            project = projects[p].metadata.name;
+                            break;
+                        }
+                        if (!project)
+                            return false; // the user has no projects
+                    }
+
+                    var request = {
+                        kind: "SubjectAccessReview",
+                        apiVersion: "v1",
+                        namespace: project,
+                        verb: verb,
+                        resource: resource
+                    };
+
+                    methods.post(loader.resolve("subjectaccessreviews"), request)
+                        .then(function(response) {
+                            sarCache[key] = response.allowed;
+                            // tell UI to update only if allowed, as our "no lookup yet" default is false already
+                            if (response.allowed) {
+                                $rootScope.$applyAsync();
```

Comment:
I found an example [how to bind ng-show to a promise](http://jsfiddle.net/BinaryMuse/ATyK3/); that fiddle uses angular 1.1.4, where this still actually works fine. But with our angular 1.3.14 it doesn't any more (test with updating the URL in "External resources"). It works fine to just update the scope property and call `$applyAsync()`, but I'll continue to search for how to do promise bindings with recent Angulars - it would be weird if that wouldn't work somehow.